modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/upgrade/v1_1_0/UpgradeImageTypeContent.java

```diff
@@ -125,57 +130,43 @@ protected String convertTypeImageElements(
 	protected void copyJournalArticleImagesToJournalRepository()
 		throws Exception {
 
-		try (LoggingTimer loggingTimer = new LoggingTimer();
-			PreparedStatement ps1 = connection.prepareStatement(
-				"select articleId, articleImageId, groupId from " +
-					"JournalArticleImage");
-			ResultSet rs1 = ps1.executeQuery()) {
-
-			while (rs1.next()) {
-				String articleId = rs1.getString(1);
-				long articleImageId = rs1.getLong(2);
-				long groupId = rs1.getLong(3);
+		StringBundler sb = new StringBundler(6);
 
-				try (PreparedStatement ps2 = connection.prepareStatement(
-						"select resourcePrimKey, userId from JournalArticle " +
-							"where groupId = ? and articleId = ?")) {
+		sb.append("select JournalArticleImage.articleImageId, ");
+		sb.append("JournalArticleImage.groupId, ");
+		sb.append("JournalArticle.resourcePrimKey, JournalArticle.userId ");
+		sb.append("from JournalArticleImage left join JournalArticle on ");
+		sb.append("(JournalArticle.groupId=JournalArticleImage.groupId) and ");
+		sb.append("(JournalArticle.articleId=JournalArticleImage.articleId)");
 
-					ps2.setLong(1, groupId);
-					ps2.setString(2, articleId);
+		List<SaveImageFileEntryCallable> saveImageFileEntryCallables =
+			new ArrayList<>();
 
-					ResultSet rs2 = ps2.executeQuery();
-
-					if (!rs2.next()) {
-						continue;
-					}
+		try (LoggingTimer loggingTimer = new LoggingTimer();
+			Statement statement = connection.createStatement();
+			ResultSet rs1 = statement.executeQuery(sb.toString())) {
 
-					long resourcePrimKey = rs2.getLong(1);
-					long userId = rs2.getLong(2);
+			while (rs1.next()) {
+				long articleImageId = rs1.getLong(1);
+				long groupId = rs1.getLong(2);
+				long resourcePrimKey = rs1.getLong(3);
+				long userId = rs1.getLong(4);
 
-					long folderId = getFolderId(
-						userId, groupId, resourcePrimKey);
+				long folderId = getFolderId(userId, groupId, resourcePrimKey);
 
-					FileEntry fileEntry =
-						PortletFileRepositoryUtil.fetchPortletFileEntry(
-							groupId, folderId, String.valueOf(articleImageId));
+				SaveImageFileEntryCallable saveImageFileEntryCallable =
+					new SaveImageFileEntryCallable(
+						articleImageId, folderId, groupId, resourcePrimKey,
+						userId);
 
-					if (fileEntry != null) {
-						continue;
-					}
+				saveImageFileEntryCallables.add(saveImageFileEntryCallable);
+			}
 
-					Image image = _imageLocalService.getImage(articleImageId);
+			ExecutorService executorService = Executors.newWorkStealingPool();
 
-					if (image == null) {
-						continue;
-					}
+			executorService.invokeAll(saveImageFileEntryCallables);
```

Comment:
@pavel-savinov @achaparro 

We are ignoring any exception thrown inside SaveImageFileEntryCallable, so in case of exception we are changing current behavior: 
 - now the upgrade is aborted
 - with this changes, in case of exceptions, the execution will continue